app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMBarrierFreeRequest.java

```diff
@@ -0,0 +1,292 @@
+package de.tum.in.tumcampusapp.tumonline;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import com.google.common.base.Optional;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.auxiliary.NetUtils;
+import de.tum.in.tumcampusapp.auxiliary.Utils;
+import de.tum.in.tumcampusapp.managers.CacheManager;
+import de.tum.in.tumcampusapp.managers.LocationManager;
+import de.tum.in.tumcampusapp.models.barrierfree.BarrierfreeContact;
+import de.tum.in.tumcampusapp.models.barrierfree.BarrierfreeMoreInfo;
+
+import static de.tum.in.tumcampusapp.tumonline.TUMRoomFinderRequest.*;
+
+/**
+ * The class used to handle Barrierfree data exchange with backend
+ */
+public class TUMBarrierFreeRequest {
```

Comment:
This should not be a separate class but rather use the TUMCabeClient in `de.tum.in.tumcampusapp.api`

Please fix this first, before I continue to review.